src/pal/src/thread/process.cpp

```diff
@@ -2718,13 +2704,20 @@ DestroyProcessModules(IN ProcessModules *listHead)
 __attribute__((destructor)) 
 void PROCNotifyProcessShutdown()
 {
-    TRACE("PROCNotifyProcessShutdown %p\n", g_shutdownCallback.RawValue());
-
+    // Call back into the coreclr to clean up the debugger transport pipes
     PSHUTDOWN_CALLBACK callback = InterlockedExchangePointer(&g_shutdownCallback, NULL);
     if (callback != NULL)
     {
         callback();
     }
+
+    // Cleanup the name continue semaphore on exit and abormal terminatation
+    sem_t *continueSem = InterlockedExchangePointer(&g_continueSem, SEM_FAILED);
+    if (continueSem != SEM_FAILED)
+    {
+        sem_close(continueSem);
+        sem_unlink(g_continueSemName);
```

Comment:
I wanted to keep the RC2 compatibility hack as simple and unobtrusive as possible. Unless I completely missed the mark there is no harm in the debuggee unlinking the name when it terminates.  The semaphore name is only for this process so when it terminates, it is fine is the it goes away. Normally by this time, the debugger has already unlinked it and in that case unlinking it again will not hurt anything.